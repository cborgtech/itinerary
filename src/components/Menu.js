import React, { Component } from "react";
import { Navbar, Nav, NavItem } from "react-bootstrap";
import { firebaseApp } from "../firebase";
import { connect } from "react-redux";
// import history from "../history";
// import { signInUser } from "../actions/index-actions";

export class Menu extends Component {
  signOut() {
    firebaseApp.auth().signOut();
    // this.props.store.dispatch(signInUser(""));
  }

  render() {
    return (
      <Navbar collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <a style={{ cursor: "pointer" }}>
              {" "}
              <strong>iTin3r@ry</strong>{" "}
            </a>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <NavItem>Signed in as: {this.props.state}</NavItem>
          </Nav>
          <Nav pullRight bsStyle="pills">
            <NavItem className="navitem" key="2" onClick={() => this.signOut()}>
              <strong>Sign Out</strong>
            </NavItem>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

function mapStateToProps(state) {
  // console.log("state", state);
  return { state };
}

export default connect(mapStateToProps, null)(Menu);
