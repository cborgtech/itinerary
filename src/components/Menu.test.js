import React from "react";
import "../setupTest.js";
import { shallow } from "enzyme";
import { Menu } from "./Menu";

describe("Menu Component", () => {
  const wrapper = shallow(<Menu />);
  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //it returns a Navbar
});
