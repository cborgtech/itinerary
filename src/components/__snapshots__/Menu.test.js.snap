// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <NavbarHeader>
          <NavbarBrand>
                    <a
                              style={
                                        Object {
                                                  "cursor": "pointer",
                                                }
                              }
                    >
                               
                              <strong>
                                        iTin3r@ry
                              </strong>
                               
                    </a>
          </NavbarBrand>
          <NavbarToggle />
</NavbarHeader>,
        <NavbarCollapse>
          <Nav
                    bsClass="nav"
                    justified={false}
                    pullLeft={false}
                    pullRight={false}
                    stacked={false}
          >
                    <NavItem
                              active={false}
                              disabled={false}
                    >
                              Signed in as: 
                    </NavItem>
          </Nav>
          <Nav
                    bsClass="nav"
                    bsStyle="pills"
                    justified={false}
                    pullLeft={false}
                    pullRight={true}
                    stacked={false}
          >
                    <NavItem
                              active={false}
                              className="navitem"
                              disabled={false}
                              onClick={[Function]}
                    >
                              <strong>
                                        Sign Out
                              </strong>
                    </NavItem>
          </Nav>
</NavbarCollapse>,
      ],
      "collapseOnSelect": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarBrand>
              <a
                            style={
                                          Object {
                                                        "cursor": "pointer",
                                                      }
                            }
              >
                             
                            <strong>
                                          iTin3r@ry
                            </strong>
                             
              </a>
</NavbarBrand>,
            <NavbarToggle />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                style={
                                Object {
                                                "cursor": "pointer",
                                              }
                }
>
                 
                <strong>
                                iTin3r@ry
                </strong>
                 
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <strong>
                    iTin3r@ry
</strong>,
                  " ",
                ],
                "style": Object {
                  "cursor": "pointer",
                },
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "iTin3r@ry",
                  },
                  "ref": null,
                  "rendered": "iTin3r@ry",
                  "type": "strong",
                },
                " ",
              ],
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Nav
              bsClass="nav"
              justified={false}
              pullLeft={false}
              pullRight={false}
              stacked={false}
>
              <NavItem
                            active={false}
                            disabled={false}
              >
                            Signed in as: 
              </NavItem>
</Nav>,
            <Nav
              bsClass="nav"
              bsStyle="pills"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
>
              <NavItem
                            active={false}
                            className="navitem"
                            disabled={false}
                            onClick={[Function]}
              >
                            <strong>
                                          Sign Out
                            </strong>
              </NavItem>
</Nav>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "children": <NavItem
                active={false}
                disabled={false}
>
                Signed in as: 
</NavItem>,
              "justified": false,
              "pullLeft": false,
              "pullRight": false,
              "stacked": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": Array [
                  "Signed in as: ",
                  undefined,
                ],
                "disabled": false,
              },
              "ref": null,
              "rendered": Array [
                "Signed in as: ",
                undefined,
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "bsStyle": "pills",
              "children": <NavItem
                active={false}
                className="navitem"
                disabled={false}
                onClick={[Function]}
>
                <strong>
                                Sign Out
                </strong>
</NavItem>,
              "justified": false,
              "pullLeft": false,
              "pullRight": true,
              "stacked": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": <strong>
                  Sign Out
</strong>,
                "className": "navitem",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Out",
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": "strong",
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
                        <a
                                    style={
                                                Object {
                                                            "cursor": "pointer",
                                                          }
                                    }
                        >
                                     
                                    <strong>
                                                iTin3r@ry
                                    </strong>
                                     
                        </a>
            </NavbarBrand>
            <NavbarToggle />
</NavbarHeader>,
          <NavbarCollapse>
            <Nav
                        bsClass="nav"
                        justified={false}
                        pullLeft={false}
                        pullRight={false}
                        stacked={false}
            >
                        <NavItem
                                    active={false}
                                    disabled={false}
                        >
                                    Signed in as: 
                        </NavItem>
            </Nav>
            <Nav
                        bsClass="nav"
                        bsStyle="pills"
                        justified={false}
                        pullLeft={false}
                        pullRight={true}
                        stacked={false}
            >
                        <NavItem
                                    active={false}
                                    className="navitem"
                                    disabled={false}
                                    onClick={[Function]}
                        >
                                    <strong>
                                                Sign Out
                                    </strong>
                        </NavItem>
            </Nav>
</NavbarCollapse>,
        ],
        "collapseOnSelect": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarBrand>
                <a
                                style={
                                                Object {
                                                                "cursor": "pointer",
                                                              }
                                }
                >
                                 
                                <strong>
                                                iTin3r@ry
                                </strong>
                                 
                </a>
</NavbarBrand>,
              <NavbarToggle />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  style={
                                    Object {
                                                      "cursor": "pointer",
                                                    }
                  }
>
                   
                  <strong>
                                    iTin3r@ry
                  </strong>
                   
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <strong>
                      iTin3r@ry
</strong>,
                    " ",
                  ],
                  "style": Object {
                    "cursor": "pointer",
                  },
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "iTin3r@ry",
                    },
                    "ref": null,
                    "rendered": "iTin3r@ry",
                    "type": "strong",
                  },
                  " ",
                ],
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Nav
                bsClass="nav"
                justified={false}
                pullLeft={false}
                pullRight={false}
                stacked={false}
>
                <NavItem
                                active={false}
                                disabled={false}
                >
                                Signed in as: 
                </NavItem>
</Nav>,
              <Nav
                bsClass="nav"
                bsStyle="pills"
                justified={false}
                pullLeft={false}
                pullRight={true}
                stacked={false}
>
                <NavItem
                                active={false}
                                className="navitem"
                                disabled={false}
                                onClick={[Function]}
                >
                                <strong>
                                                Sign Out
                                </strong>
                </NavItem>
</Nav>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "nav",
                "children": <NavItem
                  active={false}
                  disabled={false}
>
                  Signed in as: 
</NavItem>,
                "justified": false,
                "pullLeft": false,
                "pullRight": false,
                "stacked": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": Array [
                    "Signed in as: ",
                    undefined,
                  ],
                  "disabled": false,
                },
                "ref": null,
                "rendered": Array [
                  "Signed in as: ",
                  undefined,
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "nav",
                "bsStyle": "pills",
                "children": <NavItem
                  active={false}
                  className="navitem"
                  disabled={false}
                  onClick={[Function]}
>
                  <strong>
                                    Sign Out
                  </strong>
</NavItem>,
                "justified": false,
                "pullLeft": false,
                "pullRight": true,
                "stacked": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": <strong>
                    Sign Out
</strong>,
                  "className": "navitem",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Out",
                  },
                  "ref": null,
                  "rendered": "Sign Out",
                  "type": "strong",
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
